pkg/registry/resource/resourceclaimtemplate/strategy_test.go

```diff
@@ -808,6 +818,39 @@ func TestStrategyUpdate(t *testing.T) {
 				}
 			},
 		},
+		"keep-existing-fields-consumable-capacity": {
+			oldObj:             objWithCapacityRequests,
+			newObj:             objWithCapacityRequests,
+			consumableCapacity: true,
+			expectObj:          objWithCapacityRequests,
+			verify: func(t *testing.T, as []testclient.Action) {
+				if len(as) != 0 {
+					t.Errorf("expected no action to be taken")
+				}
+			},
+		},
+		"keep-existing-fields-consumable-capacity-disabled-feature": {
+			oldObj:             objWithCapacityRequests,
+			newObj:             objWithCapacityRequests,
+			consumableCapacity: false,
+			expectObj:          objWithCapacityRequests,
+			verify: func(t *testing.T, as []testclient.Action) {
+				if len(as) != 0 {
+					t.Errorf("expected no action to be taken")
+				}
+			},
+		},
+		"drop-fields-consumable-capacity": {
+			oldObj:                 obj,
+			newObj:                 objWithCapacityRequests,
+			consumableCapacity:     false,
+			expectValidationErrors: []string{fieldImmutableError}, // Spec is immutable.
```

Comment:
... and after dropping the `DistinctAttribute` field there's still one additional constraint.

Right?

Reply:
@pohly Yes, it is a different issue here. I add the test case where the previous object doesn't contain the field and the new object contains. The ResourceClaimTemplate is different from ResourceClaim where I can test the drop field use case in update strategy by adding new allocation in status. ResourceClaimTemplate contains only spec and it is not allowed to change at the first place due to the immutable error. 
Meanwhile, this new object will have no issue (without DistinctAttribute) to be dropped when come to the `TestClaimTemplateStrategyCreate`.